### PR TITLE
[libc] Fix startup utilities failing to install in full build mode

### PR DESCRIPTION
**Pull request by jhuber6**
Summary:
Currently, doing `ninja install` will fail in fullbuild mode due to the
startup utilities not being built by default. This was hidden previously
by the fact that if tests were run, it would build the startup utilities
and thus they would be present.

This patch solves this issue by making the `libc-startup` target a
dependncy on the final library. Furthermore we simply factor out the
library install directory into the base CMake directory next to the
include directory handling. This change makes the `crt` files get
installed in `lib/x86_64-unknown-linu-gnu` instead of just `lib`.

This fixes an error I had where doing a runtimes failed to install its
libraries because the install step always errored.

### CODE REVIEW

**Comment by jhuber6** [comment]
Also realized that actually installing the startup utilities breaks compiling with the compiler. I remember we previously hid these behind `install-libc` or something. So we likely want to move the installation off of the standard target altogether.

File: libc/CMakeLists.txt [hunk 1]
```diff
@@ -225,6 +225,15 @@ else()
   set(LIBC_INSTALL_INCLUDE_DIR ${CMAKE_INSTALL_INCLUDEDIR})
 endif()
 
+if(LIBC_TARGET_TRIPLE)
```
Comment (SchrodingerZhu) [inline libc/CMakeLists.txt:228]: Can you comment a bit about the logic where (i.e. why checking TARGET_TRIPLE)

Comment (jhuber6) [reply libc/CMakeLists.txt:228]: That's the existing logic  that's been moved, so it's not related to the patch. But `LIBC_TARGET_TRIPLE` is set whenever building using https://libc.llvm.org/full_cross_build.html#runtimes-cross-build.

Comment (jhuber6) [reply libc/CMakeLists.txt:228]: The intention is to install these things in subdirectories, like `libcxx` and `libc` end up in `lib/x86_64-unknown-linux-gnu`.

File: libc/lib/CMakeLists.txt [hunk 2]
```diff
@@ -35,19 +35,11 @@ foreach(archive IN ZIP_LISTS
   )
   if(LLVM_LIBC_FULL_BUILD)
     target_link_libraries(${archive_1} PUBLIC libc-headers)
+    add_dependencies(${archive_1} libc-startup)
```
Comment (michaelrj-google) [inline libc/lib/CMakeLists.txt:38]: are there any fullbuild targets we support that don't have startup code yet? (E.g. risc-v linux?)

Comment (jhuber6) [reply libc/lib/CMakeLists.txt:38]: I think `linux` and `gpu` are the only ones with startup code currently. But good catch, should be `if TARGET`.

**Review by michaelrj-google — APPROVED** [review]
LGTM, hopefully we can eventually remove this from `EXCLUDE_FROM_ALL` but that will probably take a while

**Pull request merged**